tools/pipelines/templates/build-npm-client-package.yml

```diff
@@ -0,0 +1,898 @@
+                    if [ `cat git_status.log | wc -l` != "0" ]; then
+                      cat git_status.log
+                      echo "##vso[task.logissue type=error]Build should not create extraneous files"
+                      exit -1;
+                    fi
+
+              # This additional build step is used to run step not part of the main build. In build client pipeline,
+              # this is used to inject telemetry key file and upload built devtools extension.
+              - ${{ parameters.additionalBuildSteps }}
+
+            templateContext:
+              outputParentDirectory: $(Build.ArtifactStagingDirectory)
+              outputs:
+                - output: pipelineArtifact
+                  displayName: Publish Artifact - nested_lib_dist
+                  targetPath: $(Build.ArtifactStagingDirectory)/nested_lib_dist
+                  artifactName: nested_lib_dist
+                  publishLocation: pipeline
+
+                - ${{ if ne(parameters.taskPack, false) }}:
+                    - output: pipelineArtifact
+                      displayName: Publish Artifact - pack
+                      targetPath: $(Build.ArtifactStagingDirectory)/pack
+                      artifactName: pack
+                      publishLocation: pipeline
+
+                    - output: pipelineArtifact
+                      displayName: Publish Artifact - Test Files
+                      targetPath: $(Build.ArtifactStagingDirectory)/test-files
+                      artifactName: test-files
+                      publishLocation: pipeline
+                      sbomEnabled: false
+
+                - ${{ if eq(parameters.taskBundleAnalysis, true) }}:
+                    - output: pipelineArtifact
+                      displayName: Publish Artifacts - bundle-analysis
+                      condition: and( succeeded(), ne(variables['Build.Reason'], 'PullRequest'), eq(${{ parameters.taskPublishBundleSizeArtifacts }}, true) )
+                      targetPath: $(Build.ArtifactStagingDirectory)/bundleAnalysis
+                      artifactName: bundleAnalysis
+                      sbomEnabled: false
+                      publishLocation: pipeline
+
+                - ${{ if or(eq(parameters.publishDocs, true), eq(parameters.taskBuildDocs, true)) }}:
+                    - output: pipelineArtifact
+                      displayName: Publish Artifact - _api-extractor-temp
+                      targetPath: $(Build.ArtifactStagingDirectory)/_api-extractor-temp
+                      artifactName: _api-extractor-temp
+                      sbomEnabled: false
+                      publishLocation: pipeline
+
+          - job: Coverage_tests
+            displayName: "Coverage tests"
+            dependsOn: build
+            variables:
+              - group: ado-feeds
+              - group: storage-vars
+              - name: releaseBuildVar
+                value: '$[variables.releaseBuild]'
```

Comment:
I'm almost certain none of these are needed in the new jobs.
- As best as I can tell, `releaseBuildVar` is not actually used anywhere. I'd suggest we don't add it in the new jobs so it doesn't introduce noise (same for the jobs below).
- The variables inside `storage-vars` are related to publishing the public website, not relevant to running tests.
- The variables inside `ado-feeds` have the URLs for some of our internal feeds, which we need when we need to install packages from them (like when running e2e tests). I don't think we do any of that here, so it can also go away.